app/jobs/repository_import_event_job.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+# Documentation: https://developer.github.com/v3/activity/events/types/#repositoryimportevent
+class RepositoryImportEventJob < ApplicationJob
+  queue_as :porter_status
+
+  CREATE_COMPLETE = "Your GitHub repository was created."
+  IMPORT_FAILED = "We were not able to import starter code to your assignment, please try again."
+
+  def perform(payload_body)
+    github_repo_id = payload_body.dig("repository", "id")
+    status = payload_body.dig("status")
+    html_url = payload_body.dig("repository", "html_url")
+
+    repo = AssignmentRepo.find_by(github_repo_id: github_repo_id)
+    return if repo.blank?
+    # Group assignments coming in a follow up PR
+    # repo ||= GroupAssignmentRepo.find_by(github_repo_id: github_repo_id)
+
+    handle_assignment_repo(repo, status, html_url)
+  end
+
+  private
+
+  # rubocop:disable MethodLength
+  def handle_assignment_repo(assignment_repo, status, html_url)
+    user = assignment_repo.user
+    invitation = assignment_repo.assignment.invitation
+    invite_status = invitation.status(user)
+
+    case status
+    when "success"
+      invite_status.completed!
+      broadcast_assignment_repo_success(user, invite_status, html_url)
+      GitHubClassroom.statsd.increment("v3_exercise_repo.import.success")
+    when "failure"
+      invite_status.errored_importing_starter_code!
+      broadcast_assignment_repo_failure(user, invite_status, html_url)
+      GitHubClassroom.statsd.increment("v3_exercise_repo.import.failure")
```

Comment:
In the case of failure here I am not destroying the Assignment repo (It's destroyed on manual retry). Should we destroy it here too?

Reply:
It's probably ok :+1: